pyganja/script_api.py

```diff
@@ -66,127 +64,116 @@ def generate_notebook_js(scene, sig=None, *,
-        // Explicitly clearing it makes ganja.js fall back to RequireJS.
-        let module = undefined;
-        """
-        js += read_ganja()
-        js += """
-        // take a closure on element before the next cell replaces it
-        (function(element) {
-            (requirejs||require)(['Algebra'], function(Algebra) {
-                var opts = """ + json.dumps(opts) + """;  // injected from python
-                var output = Algebra({p: opts.p, q: opts.q, r: opts.r, baseType: Float64Array}).inline((opts)=>{
-                    var data = """ + script_json + """;  // injected from python
-
-                    // When we get a file, we load and display.
-                    var canvas;
-                    var h=0, p=0;
-                    // convert arrays of floats back to CGA elements.
-                    data = data.map(x=>x.length==opts.mv_length?new Element(x):x);
-                    // add the graph to the page.
-                    canvas = this.graph(data, opts.graph);
-                    canvas.options.h = h;
-                    canvas.options.p = p;
-                    // make it big.
-                    canvas.style.width = '100%';
-                    canvas.style.height = '50vh';
-                    return canvas;
-                })(opts);
-                element.append(output);
-
-                var a = document.createElement("button");
-                var t = document.createTextNode("\N{FLOPPY DISK} Save");
-                a.appendChild(t);
-                function screenshot(){
-                    //output.width = 1920;  output.height = 1080;
-                    output.update(output.value);
-                    output.toBlob(function(blob) {
-                        var url = URL.createObjectURL(blob);
-                        window.open(url, '_blank');
-                    });
-                }
-                window.addEventListener('resize', function() {
-                    output.update(output.value);
+    )
+    if p - q == 2:  # 2D
+        opts['graph']['gl'] = False
+    return opts
+
+
+def generate_ganja_script(script_json, opts, height_str, width_str):
+    return """
+    var opts = """ + json.dumps(opts) + """;  // injected from python
+    var output = Algebra({p: opts.p, q: opts.q, r: opts.r, baseType: Float64Array}).inline((opts)=>{
+        var data = """ + script_json + """;  // injected from python
+
+        // When we get a file, we load and display.
+        var canvas;
+        var h=0, p=0;
+        // convert arrays of floats back to CGA elements.
+        data = data.map(x=>x.length==opts.mv_length?new Element(x):x);
+        // add the graph to the page.
+        canvas = this.graph(data, opts.graph);
+        canvas.options.h = h;
+        canvas.options.p = p;
+        // make it big.
+        canvas.style.width = """ + width_str + """;
+        canvas.style.height = """ + height_str + """;
```

Comment:
```suggestion
        canvas.style.width = """ + json.dumps(width_str) + """;
        canvas.style.height = """ + json.dumps(height_str) + """;
```
Saves you with messing with escaping quotes at the caller